koku/api/report/queries.py

```diff
@@ -359,7 +325,7 @@ def _get_group_by(self):
         # For that ranking to work we can't also group by instance_type.
         inherent_group_by = self._mapper._report_type_map.get('group_by')
         if (inherent_group_by and not (group_by and self._limit)):
-            group_by += inherent_group_by
+            group_by = group_by + list(set(inherent_group_by) - set(group_by))
```

Comment:
This is a bugfix. 

The refactor exposed an issue where `list += other_list` was adding duplicate entries to the merged result. This was causing problems with the report data nesting. So, this ensures we de-duplicate the lists.